microsoft-edge/webview2/concepts/threading-model.md

```diff
@@ -1,85 +1,133 @@
+{
+   // Post web message when button is clicked
+   this.webView2Control.ExecuteScriptAsync("window.chrome.webview.postMessage(\"Open Dialog\");");
+}
+
+private void CoreWebView2_WebMessageReceived(object sender, CoreWebView2WebMessageReceivedEventArgs e)
+{
+   string msg = e.TryGetWebMessageAsString();
+   if (msg == "Open Dialog")
+   {
+      Form1 form = new Form1(); // Create a new form that contains a new WebView when web message is received.
+      form.ShowDialog(); // This will cause a reentrancy issue and cause the newly created WebView inside the modal dialog to hang.
+   }
+}
+```     
+
+Instead, schedule the appropriate work to take place after completion of the event handler, as shown in the following code.
+
+```csharp
+private void CoreWebView2_WebMessageReceived(object sender, CoreWebView2WebMessageReceivedEventArgs e)
+{
+   string msg = e.TryGetWebMessageAsString();
+   if (msg == "Open Dialog")
+   {
+      // Show a modal dialog after the current event handler is completed, to avoid potential reentrancy caused by running a nested message loop in the WebView2 event handler.
+      System.Threading.SynchronizationContext.Current.Post((_) => {
+         Form1 form = new Form1(); 
+         form.ShowDialog();
+         form.Closed();
+      }, null);
+   }
+}
+``` 
+
+> [!NOTE]
+> For WinForms and WPF apps, to get the full call stack for debugging purposes, you must turn on native code debugging for WebView2 apps, as follows.
+> 1.  Open your WebView2 project in Visual Studio.
+> 1.  In **Solution Explorer**, right-click the WebView2 project and then select **Properties**.  
+> 1.  Select the **Debug** tab, and then select the **Enable native code debugging** checkbox, as shown below.
+
+:::image type="complex" source="../media/webview-enable-native-debug.png" alt-text="Enabling native code debugging in Visual Studio" lightbox="../media/webview-enable-native-debug.png":::
+   Enabling native code debugging in Visual Studio
+:::image-end:::  
+
+## Deferrals  
+
+Some WebView2 events read values that are set on the related event arguments, or start some action after the event handler completes.  If you also need to run an asynchronous operation, such as an event handler, use the `GetDeferral` method on the event arguments of the associated events.  The returned `Deferral` object ensures that the event handler isn't considered complete until the `Complete` method of the `Deferral` is requested.  
+
+For instance, you can use the `NewWindowRequested` event to provide a `CoreWebView2` to connect as a child window when the event handler completes.  But if you need to asynchronously create the `CoreWebView2`, you should request the `GetDeferral` method on the `NewWindowRequestedEventArgs`.  After you've asynchronously created the `CoreWebView2` and set the `NewWindow` property on the `NewWindowRequestedEventArgs`, request `Complete` on the `Deferral` object that's returned by the `GetDeferral` method.  

```

Comment:
(re-entering this comment b/c not sure of github UI)
Please check how I reworded the end of the paragraph, instead of "object be returned".
@tofuandeve @jasonstephen15

Reply:
Looks good to me!